sys/vm/vm_mmap.c

```diff
@@ -1459,6 +1459,20 @@ kern_mincore(struct thread *td, uintptr_t addr0, size_t len,
 				    pmap_is_referenced(m) ||
 				    (m->a.flags & PGA_REFERENCED) != 0)
 					mincoreinfo |= MINCORE_REFERENCED_OTHER;
+
+				/*
+				 * The pmap may have told us that this
+				 * particular mapping is already capdirty,
+				 * and if so don't bother with the more
+				 * elaborate probes and especially not with
+				 * the expensive pmap_is_capdirty.
+				 */
+				if ((mincoreinfo & MINCORE_CAPDIRTY) == 0 &&
+				    ((m->a.flags & PGA_CAPDIRTY) != 0 ||
+				    (m->flags & VPO_FASTCAPDIRTY) != 0 ||
+				    pmap_is_capdirty(m) ||
+				    (m->a.flags & PGA_CAPDIRTY) != 0))
```

Comment:
Just want to confirm that pmap_is_capdirty() changes the value of m->a.flags because the condition is checked twice.

Reply:
`pmap_is_capdirty` takes the `pmap` lock and so is an opportunity for contention, even if it doesn't explicitly change the flag.  This was much more important when `mincore` was going to be how we did efficient page skipping in userland, so I've just dropped this particular bit of complexity for this PR.